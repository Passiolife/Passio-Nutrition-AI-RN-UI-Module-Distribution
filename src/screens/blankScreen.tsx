import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { WidgetsCard } from '../components';
import { ICONS } from '../assets';

export const BlankScreen = () => {
  return (
    <View style={styles.bodyContainer}>
      <Text>WIP</Text>
      <WidgetsCard
        widgetTitle="Water"
        leftIcon={ICONS.blueWaterDrop}
        rightIcon={ICONS.newAddPlus}
        value={45}
        unitValue="OZ"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  bodyContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
});
