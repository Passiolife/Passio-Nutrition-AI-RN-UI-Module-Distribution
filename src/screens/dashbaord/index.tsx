export * from './DashboardScreen';
export * from './DashbaordScreen.Style';
export * from './utils';
