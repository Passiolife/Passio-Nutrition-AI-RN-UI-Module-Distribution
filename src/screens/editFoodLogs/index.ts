export * from './editFoodLogsScreen';
export * from './alerts';
export * from './utils';
