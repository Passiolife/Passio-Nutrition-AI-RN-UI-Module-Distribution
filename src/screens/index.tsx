export * from './nutritionProfile';
export * from './quick';
export * from './editFoodLogs';
export * from './scanning';
export * from '../models/FoodLog';
export * from '../models/FavoriteFoodItem';
export * from '../NutritionApp';
export * from './dashbaord';
export * from './meallogss';
export * from './foodSearch';
export * from './myFavoritess';
export * from './editIngredients';
export * from './recipeEditor';
export * from './onboardings';
export * from './myPlans';
export * from './home';
export * from './water';
export * from './weight';
export * from './nutritionInformation';
export * from './voiceLogging';
export * from './takePicture';
export * from './advisor';
export * from './imagePicker';
export * from './foodCreator';
export * from './myFoods';
export * from './barcode';
export * from './editRecipe';
