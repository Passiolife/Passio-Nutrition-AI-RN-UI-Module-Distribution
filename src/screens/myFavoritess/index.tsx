import MyFavoritesScreen from './MyFavoritesScreen';
export { MyFavoritesScreen };
export * from './MyFavoritesScreen';
export * from './views';
