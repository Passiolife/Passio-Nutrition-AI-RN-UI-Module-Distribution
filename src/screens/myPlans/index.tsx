export * from './MyPlanScreen';
export * from './mealPlan';
export * from './suggestions';
