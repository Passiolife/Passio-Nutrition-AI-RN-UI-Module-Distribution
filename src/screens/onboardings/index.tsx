import OnboardingScreen from './OnboardingScreen';
export { OnboardingScreen };
