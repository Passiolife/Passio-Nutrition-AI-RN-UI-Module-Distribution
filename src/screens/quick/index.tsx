export * from './views';
export * from './QuickScanningScreen';
