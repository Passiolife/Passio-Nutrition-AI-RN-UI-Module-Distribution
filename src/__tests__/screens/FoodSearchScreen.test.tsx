// @ts-nocheck
import { type RenderAPI, render } from '@testing-library/react-native';

import { FoodSearchScreen } from '../../screens';
import { MockProviders } from '../provider/MockProviders';
import type { PassioIDAttributes } from '@passiolife/nutritionai-react-native-sdk-v3/src/sdk/v2';
import React from 'react';

const mockedDispatch = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
    useRoute: () => ({
      params: {
        logToDate: { undefined },
        logToMeal: { undefined },
      },
    }),
  };
});
jest.mock('@passiolife/nutritionai-react-native-sdk-v3', () => {
  const mockSDK = {
    getAttributesForPassioID(): Promise<PassioIDAttributes | null> {
      return require('../assets/json/mock_passio_sdk_attribute.json');
    },
  };
  const mockPassioIcon = require('../utils/MockPassioIcon').default;

  return {
    PassioSDK: mockSDK,
    PassioIconView: mockPassioIcon,
    IconSize: 'PX90',
    PassioIDEntityType: {
      group: 'group',
    },
  };
});

describe('search view', () => {
  it('Renders correctly with no data', () => {
    let tree: RenderAPI;
    tree = render(
      <>
        <MockProviders>
          <FoodSearchScreen
            logToDate={undefined}
            logToMeal={undefined}
            defaultPassioIdAttributes={[]}
            isLoading={false}
          />
        </MockProviders>
      </>
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly with search results', async () => {
    const passioResult: PassioIDAttributes[] = require('../assets/json/search_result.json');
    let tree: RenderAPI;
    tree = render(
      <>
        <MockProviders>
          <FoodSearchScreen
            logToDate={undefined}
            logToMeal={undefined}
            defaultPassioIdAttributes={passioResult}
            isLoading={false}
          />
        </MockProviders>
      </>
    );
    expect(tree).toMatchSnapshot();
  });
});
