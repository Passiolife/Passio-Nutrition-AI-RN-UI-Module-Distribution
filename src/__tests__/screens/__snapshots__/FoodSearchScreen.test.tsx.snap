// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search view Renders correctly with no data 1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#286CE2",
        "height": 20,
      }
    }
  />,
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#286CE2",
              "justifyContent": "space-between",
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            },
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 12,
                "flex": 1,
                "marginEnd": 16,
                "paddingHorizontal": 10,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              {
                "testUri": "../../../src/assets/ic_search_gray_2x.png",
              }
            }
            style={
              {
                "height": 25,
                "width": 25,
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "white",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            <View>
              <TextInput
                onChangeText={[Function]}
                placeholder="search here"
                style={
                  [
                    {
                      "backgroundColor": "#EFF0F6",
                      "borderColor": "#D9DBE9",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "height": 40,
                      "paddingHorizontal": 8,
                    },
                    {},
                    {
                      "backgroundColor": "white",
                      "borderRadius": 0,
                      "borderWidth": 0,
                      "fontSize": 16,
                    },
                  ]
                }
                testID="testSearchInput"
                value=""
              />
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#286CE2",
                "borderRadius": 50,
                "height": 22,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 5,
                "width": 22,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../src/assets/ic_close_white.png",
                }
              }
              style={
                {
                  "height": 10,
                  "width": 10,
                }
              }
            />
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "paddingVertical": 10,
              }
            }
          >
            Cancel
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#003389",
            "fontSize": 24,
            "fontWeight": "400",
            "marginVertical": 20,
            "textAlign": "center",
          }
        }
      >
        Recent
      </Text>
      <RCTScrollView
        ListEmptyComponent={[Function]}
        contentContainerStyle={
          {
            "paddingBottom": 25,
          }
        }
        data={[]}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={0.0001}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "height": 1244,
                "justifyContent": "center",
              }
            }
          >
            <Text>
              No recently logged foods
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;

exports[`search view Renders correctly with search results 1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#286CE2",
        "height": 20,
      }
    }
  />,
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#286CE2",
              "justifyContent": "space-between",
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            },
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 12,
                "flex": 1,
                "marginEnd": 16,
                "paddingHorizontal": 10,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              {
                "testUri": "../../../src/assets/ic_search_gray_2x.png",
              }
            }
            style={
              {
                "height": 25,
                "width": 25,
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "white",
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            <View>
              <TextInput
                onChangeText={[Function]}
                placeholder="search here"
                style={
                  [
                    {
                      "backgroundColor": "#EFF0F6",
                      "borderColor": "#D9DBE9",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "height": 40,
                      "paddingHorizontal": 8,
                    },
                    {},
                    {
                      "backgroundColor": "white",
                      "borderRadius": 0,
                      "borderWidth": 0,
                      "fontSize": 16,
                    },
                  ]
                }
                testID="testSearchInput"
                value=""
              />
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#286CE2",
                "borderRadius": 50,
                "height": 22,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 5,
                "width": 22,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../src/assets/ic_close_white.png",
                }
              }
              style={
                {
                  "height": 10,
                  "width": 10,
                }
              }
            />
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "paddingVertical": 10,
              }
            }
          >
            Cancel
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#003389",
            "fontSize": 24,
            "fontWeight": "400",
            "marginVertical": 20,
            "textAlign": "center",
          }
        }
      >
        Recent
      </Text>
      <RCTScrollView
        ListEmptyComponent={[Function]}
        contentContainerStyle={
          {
            "paddingBottom": 25,
          }
        }
        data={[]}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={0.0001}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "height": 1244,
                "justifyContent": "center",
              }
            }
          >
            <Text>
              No recently logged foods
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;
